tests/interlocked_int/interlocked_int.c

```diff
@@ -0,0 +1,855 @@
+
+    ///assert
+    ASSERT_ARE_EQUAL(uint32_t, 0, destination);
+    ASSERT_ARE_EQUAL(uint32_t, 0xF0F0F0F0, return_val);
+}
+
+/*
+Tests_SRS_INTERLOCKED_43_003 [ interlocked_and_16 shall perform an atomic bitwise AND operation on the 16-bit integer values *destination and value and store the result in *destination.]
+Tests_SRS_INTERLOCKED_43_034: [ interlocked_and_16 shall return the initial value of *destination.]
+*/
+
+TEST_FUNCTION(interlocked_and_16_does_bitwise_and)
+{
+    ///arrange
+    volatile uint16_t destination = 0xF0F0;
+    uint16_t value = 0x0F0F;
+
+    ///act
+    uint16_t return_val = (uint16_t)interlocked_and_16((volatile int16_t*)&destination, (int16_t)value);
+
+
+    ///assert
+    ASSERT_ARE_EQUAL(uint16_t, 0, destination);
+    ASSERT_ARE_EQUAL(uint16_t, 0xF0F0, return_val);
+}
+
+/*
+Tests_SRS_INTERLOCKED_43_004 [ interlocked_and_64 shall perform an atomic bitwise AND operation on the 64-bit integer values *destination and value and store the result in *destination.]
+Tests_SRS_INTERLOCKED_43_035: [ interlocked_and_64 shall return the initial value of *destination.]
+*/
+
+TEST_FUNCTION(interlocked_and_64_does_bitwise_and)
+{
+    ///arrange
+    volatile uint64_t destination = 0xF0F0F0F0F0F0F0F0;
+    uint64_t value = 0x0F0F0F0F0F0F0F0F;
+
+    ///act
+    uint64_t return_val = (uint64_t)interlocked_and_64((volatile int64_t*)&destination, (int64_t)value);
+
+    ///assert
+    ASSERT_ARE_EQUAL(uint64_t, 0, destination);
+    ASSERT_ARE_EQUAL(uint64_t, 0xF0F0F0F0F0F0F0F0, return_val);
+}
+
+/*
+Tests_SRS_INTERLOCKED_43_005 [ interlocked_and_8 shall perform an atomic bitwise AND operation on the 8-bit integer values *destination and value and store the result in *destination]
+Tests_SRS_INTERLOCKED_43_036: [ interlocked_and_8 shall return the initial value of *destination. ]
+*/
+TEST_FUNCTION(interlocked_and_8_does_bitwise_and)
+{
+    ///arrange
+    volatile uint8_t destination = 0xF0;
+    uint8_t value = 0x0F;
+
+    ///act
+    uint8_t return_val = (uint8_t)interlocked_and_8((volatile int8_t*)&destination, (int8_t)value);
+
+    ///assert
+    ASSERT_ARE_EQUAL(uint8_t, 0, destination);
+    ASSERT_ARE_EQUAL(uint8_t, 0xF0, return_val);
+}
+
+/*
+Tests_SRS_INTERLOCKED_43_006 [ interlocked_compare_exchange shall compare the 32-bit integers pointed to by destination and comperand. If they are equal, *destination is set to exchange. These operations are performed atomically.*]
+Tests_SRS_INTERLOCKED_43_037: [ interlocked_compare_exchange shall return the initial value of *destination.]
+*/
+
+TEST_FUNCTION(interlocked_compare_exchange_equal_case)
+{
+    ///arrange
+    volatile int32_t destination = INT32_MAX;
+    int32_t comperand = INT32_MAX;
+    int32_t exchange = INT32_MIN;
+
+    ///act
+    int32_t return_val = interlocked_compare_exchange(&destination, exchange, comperand);
+
+    ///assert
+    ASSERT_ARE_EQUAL(int32_t, INT32_MIN, destination);
+    ASSERT_ARE_EQUAL(int32_t, INT32_MAX, return_val);
+}
+/*
```

Comment:
There seems to be a bit of inconsistency in the way the tests are laid out. For example there is no line between these 2, but the subsequent ones have a new line between the tests.
I can read all, but maybe you want to have a quick pass to make them neat and consistent.